pkg/scheduler/framework/cycle_state_test.go

```diff
@@ -31,44 +34,79 @@ func (f *fakeData) Clone() StateData {
 	return copy
 }
 
-func TestCycleStateClone(t *testing.T) {
-	var key StateKey = "key"
-	data1 := "value1"
-	data2 := "value2"
+var key StateKey = "key"
 
-	state := NewCycleState()
-	originalValue := &fakeData{
-		data: data1,
-	}
-	state.Write(key, originalValue)
-	stateCopy := state.Clone()
+// createCycleStateWithFakeData creates *CycleState with fakeData.
+// The given data is used in stored fakeData.
+func createCycleStateWithFakeData(data string, recordPluginMetrics bool) *CycleState {
+	c := NewCycleState()
+	c.Write(key, &fakeData{
+		data: data,
+	})
+	c.SetRecordPluginMetrics(recordPluginMetrics)
+	return c
+}
 
-	valueCopy, err := stateCopy.Read(key)
-	if err != nil {
-		t.Errorf("failed to read copied value: %v", err)
-	}
-	if v, ok := valueCopy.(*fakeData); ok && v.data != data1 {
-		t.Errorf("clone failed, got %q, expected %q", v.data, data1)
+func TestCycleStateClone(t *testing.T) {
+	tests := []struct {
+		name            string
+		state           *CycleState
+		wantClonedState *CycleState // clonedState is CycleState cloned by Clone().
+	}{
+		{
+			name:            "clone with recordPluginMetrics true",
+			state:           createCycleStateWithFakeData("data", true),
+			wantClonedState: createCycleStateWithFakeData("data", true),
+		},
+		{
+			name:            "clone with recordPluginMetrics false",
+			state:           createCycleStateWithFakeData("data", false),
+			wantClonedState: createCycleStateWithFakeData("data", false),
+		},
+		{
+			name:            "clone with nil CycleState",
+			state:           nil,
+			wantClonedState: nil,
+		},
 	}
 
-	originalValue.data = data2
-	original, err := state.Read(key)
-	if err != nil {
-		t.Errorf("failed to read original value: %v", err)
-	}
-	if v, ok := original.(*fakeData); ok && v.data != data2 {
-		t.Errorf("original value should change, got %q, expected %q", v.data, data2)
-	}
+	cmpOpts := []cmp.Option{
+		cmp.AllowUnexported(CycleState{}),
+		cmp.Comparer(func(a, b sync.Map) bool {
+			isEqual := true
+			a.Range(func(k, v1 interface{}) bool {
+				v2, ok := b.Load(k)
+				if !ok {
+					isEqual = false
+					return false
+				}
 
-	if v, ok := valueCopy.(*fakeData); ok && v.data != data1 {
-		t.Errorf("cloned copy should not change, got %q, expected %q", v.data, data1)
+				if !cmp.Equal(v1, v2, cmp.AllowUnexported(fakeData{})) {
+					isEqual = false
+					return false
+				}
+				return true
+			})
+			return isEqual
```

Comment:
also need to check if the number of elements is the same

Reply:
Definitely. Fixed.